accord-core/src/main/java/accord/coordinate/tracking/AppliedTracker.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package accord.coordinate.tracking;
+
+import java.util.Set;
+
+import accord.local.Node;
+import accord.topology.Shard;
+import accord.topology.Topologies;
+
+import static accord.coordinate.tracking.AbstractTracker.ShardOutcomes.Fail;
+import static accord.coordinate.tracking.AbstractTracker.ShardOutcomes.NoChange;
+import static accord.coordinate.tracking.AbstractTracker.ShardOutcomes.Success;
+
+public class AppliedTracker extends AbstractTracker<AppliedTracker.AppliedShardTracker>
+{
+    public static class AppliedShardTracker extends ShardTracker
+    {
+        protected int waitingOn;
+
+        public AppliedShardTracker(Shard shard, Set<Node.Id> nonParticipants)
+        {
+            super(shard);
+            waitingOn = shard.rf() - (nonParticipants.isEmpty() ? 0 : (int)shard.nodes.stream().filter(nonParticipants::contains).count());
```

Comment:
It's for when we start marking nodes as faulty/down, so that we can say a range is "globally" durable even when some nodes are offline. I can remove it for now if you like.

Reply:
for now I think its best to remove, makes it harder for me to reason this is correct

Reply:
in my feedback branch, I have a commit to remove this